homeassistant/components/sensibo/binary_sensor.py

```diff
@@ -122,32 +122,50 @@ async def async_setup_entry(
 
     coordinator = entry.runtime_data
 
-    entities: list[SensiboMotionSensor | SensiboDeviceSensor] = []
-
-    for device_id, device_data in coordinator.data.parsed.items():
-        if device_data.motion_sensors:
+    added_devices: set[str] = set()
+
+    def _add_remove_devices() -> None:
+        """Handle additions of devices and sensors."""
+
+        entities: list[SensiboMotionSensor | SensiboDeviceSensor] = []
+        _added_devices = added_devices.copy()
```

Comment:
I don't believe you have to copy

Reply:
I need a copy because I can't change the same set I'm iterating.